### PR TITLE
Compiler and clang-tidy cleanups

### PR DESCRIPTION
**Pull request by hjmjohnson**
A few code improvements are indicated by clang-tidy and compiler warnings.

### CODE REVIEW

File: src/Filtering/ImageGrid/FitSplineIntoPointSet/Code.cxx [hunk 1]
```diff
@@ -28,32 +28,26 @@ main()
...
   pointSet->SetPoint(1, param1);
   pointSet->SetPointData(1, p1);
 
-  param2[0] = 2.0;
-  DataType p2;
-  p2[0] = 180.0;
-  p2[1] = 180.0;
+
+  PointSetType::PointType param2{ 2.0 };
+  DataType                p2({ 100.0, 100.0 });
```
Comment (dzenanz) [inline src/Filtering/ImageGrid/FitSplineIntoPointSet/Code.cxx:49]: Was there some reason to change this from 180 to 100?

File: src/Core/Mesh/WriteMeshToVTP/Code.cxx [hunk 2]
```diff
@@ -48,20 +48,10 @@ CreateMeshWithEdges()
...
-  p1[0] = 1.0;
-  p1[1] = -1.0;
-  p1[2] = 0.0; // second point (  1, -1, 0 )
-  p2[0] = 1.0;
-  p2[1] = 1.0;
-  p2[2] = 0.0; // third  point (  1,  1, 0 )
-  p3[0] = 1.0;
-  p3[1] = 1.0;
-  p3[2] = 1.0; // third  point (  1,  1, 1 )
+  const MeshType::PointType p0({ -1.0, -1.0, 0 });  // first  point ( -1, -1, 0 )
```
Comment (dzenanz) [inline src/Core/Mesh/WriteMeshToVTP/Code.cxx:51]: These comments are far less useful now.

**Review by dzenanz — APPROVED** [review]
Mostly looks good

File: src/Filtering/ImageGrid/FitSplineIntoPointSet/Code.cxx (hunk 1 above)
Comment (hjmjohnson) [reply src/Filtering/ImageGrid/FitSplineIntoPointSet/Code.cxx:49]: I am blind.  I did not read the number correctly. Fix is coming shortly.

**Pull request merged**